python3/bin/perfmon

```diff
@@ -0,0 +1,1620 @@
+        log_err("Error %s in get_percent_mem_usage, return 0.0 instead" % e)
+        return 0.0
+
+
+def get_percent_sr_usage(mylist):
+    """
+    Get the percent usage of the SR.
+    Input list should be exactly two items: [physical_utilisation, size]
+    """
+    try:
+        if len(mylist) != 2:
+            raise IncorrectInputException(
+                "Incorrect number of values to consolidate: %d (exactly 2 values)"
+                % len(mylist)
+            )
+        physical_utilisation, size = mylist[0:2]
+        return float(physical_utilisation) / size
+    except Exception as e:
+        log_err("Error %s in get_percent_sr_usage, return 0.0 instead" % e)
+        return 0.0
+
+
+# pylint: disable=too-few-public-methods
+class VariableConfig:
+    """Object storing the configuration of a Variable
+
+    Initialisation parameters:
+    xmldoc = dom object representing the <variable> nodes in the ObjectMonitor config strings.
+            See VMMonitor.__doc__ and HOSTMonitor.__doc__
+    alarm_create_callback =
+            callback called by Variable.update() to create and send an alarm
+    get_default_variable_config =
+            a function that VariableConfig.__init__() uses to lookup default tag values
+            by variable name
+    """
+
+    def __init__(self, xmldoc, alarm_create_callback, get_default_variable_config):
+        try:
+            name = xmldoc.getElementsByTagName("name")[0].getAttribute("value")
+        except IndexError as e:
+            raise XmlConfigException("variable missing 'name' tag") from e
+
+        def get_value(tag):
+            try:
+                return xmldoc.getElementsByTagName(tag)[0].getAttribute("value")
+            except Exception:
+                return get_default_variable_config(name, tag)
+
+        rrd_regex = get_value("rrd_regex")
+        consolidation_fn = get_value("consolidation_fn")
+        alarm_trigger_level = get_value("alarm_trigger_level")
+        alarm_trigger_period = get_value("alarm_trigger_period")
+        alarm_auto_inhibit_period = get_value("alarm_auto_inhibit_period")
+        alarm_trigger_sense = get_value("alarm_trigger_sense")
+        alarm_priority = get_value("alarm_priority")
+
+        # Save xmldoc: we need this when creating the body of the alarms
+        self.xmldoc = xmldoc
+
+        self.name = name
+        try:
+            self.rrd_regex = re.compile("^%s$" % rrd_regex)
+        except Exception as e:
+            raise XmlConfigException(
+                "variable %s: regex %s does not compile" % (name, rrd_regex)
+            ) from e
+
+        if consolidation_fn not in supported_consolidation_functions:
+            raise XmlConfigException(
+                "variable %s: consolidation function %s not supported"
+                % (name, consolidation_fn)
+            )
+        # It's fine to use eval here
+        # pylint: disable=eval-used
+        self.consolidation_fn = eval(consolidation_fn)
```

Comment:
Could this `eval` be get rid of through a map(string -> callable) ?

Reply:
Also tracked by CP-48628